connect/transforms/src/main/java/org/apache/kafka/connect/transforms/Cast.java

```diff
@@ -102,8 +105,10 @@ public void configure(Map<String, ?> props) {
     @Override
     public R apply(R record) {
         if (operatingSchema(record) == null) {
+            log.trace("Applying Cast SMT with null schema. Record: {}", record);
```

Comment:
Would it be better to log these traces inside `TransformChain`, where we could see the progression of the records through the chain?

Reply:
My original idea was that this gives some more useful information about which branch we are taking. Given your comment and a second thought, I think we a log in the Chain since we can't count on every SMT having the right logs, and it will provide consistency.

Reply:
I will defer to your judgement here since I haven't debugged one of these in production before -- new strategy will be to put log in TransformChain.apply() for each transformation, and then put some smaller trace logs in transformations that distinguish between schema'd and schemaless. If the schema/schemaless distinction isn't very useful, I can strike that